app/src/main/java/com/movieapp/network/NetworkAPI.java

```diff
@@ -0,0 +1,50 @@
+package com.movieapp.network;
+
+import com.movieapp.BuildConfig;
+
+import java.io.IOException;
+
+import okhttp3.HttpUrl;
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class NetworkAPI {
+
+    public static String BASE_URL = "https://api.themoviedb.org/3/";
+    private static final String API_KEY = BuildConfig.API_KEY;
+
+    private static Retrofit retrofit;
+
+    private static OkHttpClient getHttpClient() {
+        OkHttpClient.Builder httpClient = new OkHttpClient.Builder();
+        httpClient.addInterceptor(new Interceptor() {
+            @Override
+            public Response intercept(Chain chain) throws IOException {
+                Request original = chain.request();
+                HttpUrl originalUrl = original.url();
+                HttpUrl url = originalUrl.newBuilder()
+                        .addQueryParameter("api_key", API_KEY)
```

Comment:
Please pass this to interceptor

Reply:
internal fun insertInterceptor() = Interceptor {
        val url = it.request().url
            .newBuilder()
            .addQueryParameter("api_key", "")
            .build()

        val newRequest = it.request()
            .newBuilder()
            .url(url)
            .build()

        it.proceed(newRequest)
    }